main_code/Email/inbox-dialogflow.py

```diff
@@ -0,0 +1,373 @@
+    emailline = str(message['payload']['headers'])
+    match = re.findall(r'[\w\.-]+@[\w\.-]+', emailline)
+    print("Message sender: " , match[1])
+    print("Message Id : ", message['id'])
+
+    return message['snippet'], match[1], message['id']
+  except errors.HttpError as error:
+    print ('An error occurred: %s' % error)
+
+
+def GetMimeMessage(service, user_id, msg_id):
+  """Get a Message and use it to create a MIME Message.
+
+  Args:
+    service: Authorized Gmail API service instance.
+    user_id: User's email address. The special value "me"
+    can be used to indicate the authenticated user.
+    msg_id: The ID of the Message required.
+
+  Returns:
+    A MIME Message, consisting of data from Message.
+  """
+  try:
+    message = service.users().messages().get(userId=user_id, id=msg_id,
+                                             format='raw').execute()
+
+
+    print(message['snippet'])
+
+    msg_str = base64.urlsafe_b64decode(message['raw'].encode('ASCII'))
+
+    mime_msg = email.message_from_string(msg_str)
+
+    return mime_msg
+  except errors.HttpError as error:
+    print ('An error occurred: %s' % error)
+
+
+def detect_intent_texts(project_id, session_id, text, language_code = 'en'):
+    """
+    project_id (string): agent specific ID
+    session_id (string): any string. indicates same conversation
+    text (string): text to process
+    language_code (string): english by default
+
+    Returns the result of detect intent with texts as inputs.
+    Using the same `session_id` between requests allows continuation
+    of the conversation."""
+
+    session_client = dialogflow.SessionsClient()
+    session = session_client.session_path(project_id, session_id)
+
+    text_input = dialogflow.types.TextInput(text=text, language_code=language_code)
+
+    query_input = dialogflow.types.QueryInput(text=text_input)
+
+    response = session_client.detect_intent(session=session, query_input=query_input)
+
+    return '{}\n'.format(response.query_result.fulfillment_text)
+
+
+def dialogflow_response(text):
+    '''text (string): message from email body
+    return (string): dialogflow response, detected task name'''
+
+    # GCP Authentication
+    os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = 'vw-2019-cse498-dfea7d9a65a3.json'
+
+    #agent on dialog flow
+    project_id = 'vw-2019-cse498'
+
+    # session_id can be any string. Specifies conversation
+    session_id = 'mock-session'
+    language_code = 'en'
+
+    response = detect_intent_texts(project_id, session_id, text, language_code)
+    return response
+
+
+def put_value_in_clipboard(value):
+    '''value (string)
+    parameter identified for task
+    no return. Copy value to clipboard'''
+    win32clipboard.OpenClipboard()
+    win32clipboard.EmptyClipboard()
+    win32clipboard.SetClipboardText(value)
+    win32clipboard.CloseClipboard()
+
+
+def get_value_from_clipboard():
+    '''returns string
+    paste from clipboard'''

```

Comment:
also add a newline after this one. Basically a new line after the function comment before code thorughout this